modmail/extensions/extension_manager.py

```diff
@@ -0,0 +1,276 @@
+# original source:
+# https://github.com/python-discord/bot/blob/a8869b4d60512b173871c886321b261cbc4acca9/bot/exts/utils/extensions.py  # noqa: E501
+# MIT License 2021 Python Discord
+import functools
+import logging
+import typing as t
+from collections import defaultdict
+from enum import Enum
+
+from discord import Colour, Embed
+from discord.ext import commands
+from discord.ext.commands import Context
+
+from modmail.bot import ModmailBot
+from modmail.log import ModmailLogger
+from modmail.utils.cogs import ModeMetadata, ModmailCog
+from modmail.utils.extensions import EXTENSIONS, unqualify
+
+log: ModmailLogger = logging.getLogger(__name__)
+
+
+BASE_PATH_LEN = __name__.count(".")
+
+EXT_METADATA = ModeMetadata(production=True, develop=True, plugin_dev=True)
+
+
+class Action(Enum):
+    """Represents an action to perform on an extension."""
+
+    # Need to be partial otherwise they are considered to be function definitions.
+    LOAD = functools.partial(ModmailBot.load_extension)
+    UNLOAD = functools.partial(ModmailBot.unload_extension)
+    RELOAD = functools.partial(ModmailBot.reload_extension)
+
+
+class ExtensionConverter(commands.Converter):
+    """
+    Fully qualify the name of an extension and ensure it exists.
+
+    The * and ** values bypass this when used with the reload command.
+    """
+
+    source_list = EXTENSIONS
```

Comment:
```suggestion
    source_list = EXTENSIONS.copy()
```

The assignment just copies the reference to the list, not the actual list, so both `source_list` and `EXTENSIONS` refer to the same list after the assignment.

Reply:
I think its intended behaviour.

Reply:
We don't want to copy this. We should keep this list as a reference, not a copy, to ensure that extensions added or removed via other methods remain in sync with the source list